data/lm/generate_lm.py

```diff
@@ -191,17 +222,37 @@ def main():
     )
     parser.add_argument(
         "--discount_fallback",
-        help="To try when such message is returned by kenlm: 'Could not calculate Kneser-Ney discounts [...] rerun with --discount_fallback'",
+        help="To try when such message is returned by kenlm: 'Could not calculate Kneser-Ney "
+             "discounts [...] rerun with --discount_fallback'",
         action="store_true",
     )
 
     args = parser.parse_args()
 
+    # Basic checks on user supplied arguments
+    if args.kenlm_bins == "path/to/kenlm/build/bin/":
+        parser.error("Update kenlm_bins from documentation value to actual path for kenlm")
+
+    if not os.path.isdir(args.kenlm_bins):
+        parser.error("kenlm_bins must be a valid directory")
+
+    kenlm_files = ["lmplz","filter","build_binary"]
+    for f in kenlm_files:
+        if not os.path.isfile(os.path.join(args.kenlm_bins, f)):
+            parser.error(f"Required kenlm file {f} not found in kenlm_bins directory")
+
+    if len(args.top_k) > len(args.input_txt):
+        parser.error("Number of top_k arguments passed should not exceed number of input_txt"
+                     " arguments passed")
+
+    if not os.path.isdir(args.output_dir):
+        parser.error("output_dir must be a valid directory")
+
     data_lower, vocab_str = convert_and_filter_topk(args)
     build_lm(args, data_lower, vocab_str)
 
     # Delete intermediate files
-    os.remove(os.path.join(args.output_dir, "lower.txt.gz"))
+    os.remove(data_lower)
```

Comment:
why do we need this?

Reply:
Was aiming for DRY (that file is defined previously, so seems safer to use the original definition again rather than restate a second time as was being done)

Reply:
Fair enough!